lib/liquid/registers/disabled_tags.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+module Liquid
+  class DisabledTags < Register
+    def initialize
+      @disabled_tags = Hash.new { |h, k| h[k] = 0 }
+    end
+
+    def disabled?(tag)
+      @disabled_tags[tag] > 0
+    end
+
+    def disable(tags)
+      tags.each { |tag| increment(tag) }
```

Comment:
Protip: you can also do `tags.each(&method(:increment))`, though it's arguably not more readable.